documentation/rosetta-modelling-component.md

```diff
@@ -662,6 +663,66 @@ E.g.:
 [DayOfWeekEnum->SAT, DayOfWeekEnum->SUN]
 ```
 
+### Data Type and Record Constructor
+
+Data types and records can be constructed using a syntax similar to JSON, i.e.,
+```
+<type name> {
+  <field name 1>: <field value 1>,
+  <field name 2>: <field value 2>,
+  etc
+}
+```
+
+As an example, consider the following data type representing an employee of a company.
+``` Haskell
+type Employee:
+  name string (1..1)
+  age int (0..1)
+  favouriteAnimal string (0..1)
+  mentor Employee (0..1)
+```
+To construct an employee called "Dwight Schrute" with a mentor called "Michael Scott" within Rosetta, the following syntax can be used.
+``` Haskell
+Employee {
+  name: "Dwight Schrute",
+  age: 42,
+  favouriteAnimal: "bear",
+  mentor: Employee {
+    name: "Michael Scott",
+    age: empty,
+    favouriteAnimal: empty,
+    mentor: empty
+  }
+}
+```
+
+#### Triple-Dot Syntax
+
+Notice that in the example above most fields of Michael Scott are actually `empty`. Types with many optional attributes is common practice in Rosetta models such as the CDM, and assigning `empty` to them explicitly can be verbose. We can solve this by using the triple-dot keyword `...`, which will implicitly assign `empty` to all absent attributes.
```

Comment:
Typo in the second sentence, not syntactically correct. "Having" missing perhaps?